snapcraft/plugins/catkin.py

```diff
@@ -247,12 +247,30 @@ def rewrite_paths(match):
                             f.write(replaced)
 
     def _finish_build(self):
-        # Fix the shebang in _setup_util.py to be portable
-        setup_util_file = os.path.join(self.rosdir, '_setup_util.py')
+        # Fix all shebangs to use the in-snap python.
+        pattern = re.compile(r'#!.*python')
+        for root, directories, files in os.walk(self.rosdir):
+            for fileName in files:
+                with open(os.path.join(root, fileName), 'r+') as f:
```

Comment:
first, the code inside the second for.
Then, maybe the two fors too, but this might be too much.